backend/src/test/java/com/bakdata/conquery/integration/tests/ApiTokenRealmTest.java

```diff
@@ -0,0 +1,218 @@
+		Response response =
+				conquery.getClient()
+						.target(HierarchyHelper.hierarchicalPath(conquery.defaultApiURIBuilder(), ApiTokenResource.class,"deleteToken"))
+						.resolveTemplate(ApiTokenResource.TOKEN, id2)
+						.request(MediaType.APPLICATION_JSON_TYPE)
+						.header("Authorization", "Bearer " + apiToken2.getToken())
+						.delete(Response.class);
+
+		assertThat(response.getStatus()).as("It is forbidden to act on ApiTokens with ApiTokens").isEqualTo(403);
+
+
+		// Delete ApiToken2 with user token
+		response =
+				conquery.getClient()
+						.target(HierarchyHelper.hierarchicalPath(conquery.defaultApiURIBuilder(), ApiTokenResource.class,"deleteToken"))
+						.resolveTemplate(ApiTokenResource.TOKEN, id2)
+						.request(MediaType.APPLICATION_JSON_TYPE)
+						.header("Authorization", "Bearer " + userToken)
+						.delete(Response.class);
+
+		assertThat(response.getStatus()).as("It is okay to act on ApiTokens with UserTokens").isEqualTo(200);
+		assertThat(realm.listUserToken(testUser)).hasSize(1);
+
+		// Try to use the deleted token to access Admin
+		response = conquery.getClient()
+				.target(HierarchyHelper.hierarchicalPath(conquery.defaultAdminURIBuilder(), AdminDatasetsResource.class,"listDatasets"))
+				.request(MediaType.APPLICATION_JSON_TYPE)
+				.header("Authorization", "Bearer " + apiToken2.getToken())
+				.get(Response.class);
+
+		assertThat(response.getStatus()).as("Cannot use deleted token").isEqualTo(401);
+
+		// Try to act on tokens from another user
+		final MetaStorage metaStorage = conquery.getMetaStorage();
+		final User user2 = new User("TestUser2", "TestUser2", metaStorage);
+		metaStorage.addUser(user2);
+		final String user2Token = conqueryTokenRealm.createTokenForUser(user2.getId());
+
+		// Try to delete ApiToken2 with ApiToken (should fail)
+		final UUID id1 = apiTokens.stream().filter(t -> t.getScopes().contains(Scopes.DATASET)).map(ApiTokenDataRepresentation.Response::getId).collect(MoreCollectors.onlyElement());
+		response =
+				conquery.getClient()
+						.target(HierarchyHelper.hierarchicalPath(conquery.defaultApiURIBuilder(), ApiTokenResource.class,"deleteToken"))
+						.resolveTemplate(ApiTokenResource.TOKEN, id1)
+						.request(MediaType.APPLICATION_JSON_TYPE)
+						.header("Authorization", "Bearer " + user2Token)
+						.delete(Response.class);
+
+		assertThat(response.getStatus()).as("It is forbidden to act on someone else ApiTokens").isEqualTo(403);
+
+		// Request ApiToken 3 (expired)
+		final ApiTokenDataRepresentation.Request tokenRequest3 = new ApiTokenDataRepresentation.Request();
+
+		tokenRequest3.setName("test-token");
+		tokenRequest3.setScopes(EnumSet.of(Scopes.DATASET));
+		tokenRequest3.setExpirationDate(LocalDate.now().minus(1, ChronoUnit.DAYS));
+
+		assertThatThrownBy(() -> requestApiToken(conquery, userToken, tokenRequest3)).as("Expiration date is in the past").isExactlyInstanceOf(ClientErrorException.class).hasMessageContaining("HTTP 422");
```

Comment:
Ok, wenn ich das richtig verstehe kann ich einfach sagen wie das Api-Token aussieht? Ist das nicht genau das was ein API-Token nciht sein sollte? Oder testest du hier nur das erzeugen eines API-Tokens? Wo testest du, dass ein API-Token verwendet wird, das schon abgelaufen ist?

Reply:
Genau, das ist hier nur ein TokenRequest.

Dass ein abgelaufenenes Token nicht mehr funktioniert teste ich in Zeile 191

Reply:
ahhh, das hatte ich übersehen. Also hier versuchst du ein abgelaufenes zu erzeugen => fail, erzeugst es dann händisch um dann damit fehlzuschlagen? Super!